RevenueCatUI/CustomerCenter/Data/PurchaseInformation.swift

```diff
@@ -123,6 +136,60 @@ struct PurchaseInformation {
 }
 // swiftlint:enable nesting
 
+extension PurchaseInformation {
+
+    /// Provides detailed information about a user's purchase, including  renewal price.
+    ///
+    /// This function fetches the renewal price details for the given product asynchronously from
+    /// StoreKit 2 and constructs a `PurchaseInformation` object with the provided
+    /// transaction, entitlement, and subscribed product details.
+    ///
+    /// - Parameters:
+    ///   - entitlement: Optional entitlement information associated with the purchase.
+    ///   - subscribedProduct: The product the user has subscribed to, represented as a `StoreProduct`.
+    ///   - transaction: The transaction information for the purchase.
+    /// - Returns: A `PurchaseInformation` object containing the purchase details, including the renewal price.
+    ///
+    @available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, visionOS 1.0, *)
+    static func purchaseInformationUsingRenewalInfo(
+        entitlement: EntitlementInfo? = nil,
+        subscribedProduct: StoreProduct,
+        transaction: Transaction,
+        customerCenterStoreKitUtilities: CustomerCenterStoreKitUtilitiesType
+    ) async -> PurchaseInformation {
+        let renewalPriceDetails = await Self.extractPriceDetailsFromRenwalInfo(
```

Comment:
Do we also need to extract the date from the `RenewalInfo`?

Reply:
I don't think we have to since we're already calculating it accurately from the transaction today, so I think we can keep doing that to keep the size of the change down. Even in the scenario where you switch product durations (say you switch from a monthly product to a yearly), the new product won't be active until the the renewal at the end of the current subscription period.

Do you know of any edge cases where the current renewal date calculation is inaccurate?